src/libstore/unix/user-lock.cc

```diff
@@ -74,8 +74,13 @@ struct SimpleUserLock : UserLock
             debug("trying user '%s'", i);
 
             struct passwd * pw = getpwnam(i.c_str());
-            if (!pw)
-                throw Error("the user '%s' in the group '%s' does not exist", i, settings.buildUsersGroup);
+            if (!pw) {
+                printError("error: the user '%s' in the group '%s' does not exist", i, settings.buildUsersGroup);
+#if __APPLE__
+                printError("note: If you are using macOS 15.0 Sequoia or newer, you may need to run this script: https://github.com/NixOS/nix/blob/master/scripts/sequoia-nixbld-user-migration.sh");
+#endif
+               continue;
```

Comment:
Is it deliberate that you're continuing instead of throwing now on Linux?

Reply:
It was deliberate because i think it is reasonable for Linux to print an error and keep trying other users, but it is worth asking/discussing if it is a good idea or not.